### ISSUE TITLE
Internal functions not properly indented

### ISSUE THREAD

**Issue by FYPetro** [affenwiesel/matlab-formatter-vscode#8]
Functions with internal functions are not properly indented. A few user preferences could come in handy here.

Before formatting:
```
function a = test(b)
a = internal(5 * b)

function a = internal(b)
a = b*internal2(b)

function a = internal2(b)
a = b*3
```

After formatting:
```
function a = test(b)
    a = internal(5 * b)

    function a = internal(b)
        a = b * internal2(b)

        function a = internal2(b)
            a = b * 3
```

Expecting:

After formatting:
```
function a = test(b)
a = internal(5 * b) % this could be indented by user's preference (main function body indent)

function a = internal(b) % this could be indented by user's preference (internal function signature indent)
    a = b * internal2(b) % this could be indented by user's preference (internal function body indent)

function a = internal2(b)
    a = b * 3 % this could be indented
```

Encoding: ANSI

**Comment by affenwiesel** [comment]
I was not even aware that this is syntactically correct matlab code... Personally I only use octave and there this code does not work.
For octave you always have to put `end `at the end of each function.
```
function a = test(b)
a = internal(5 * b)
end

function a = internal(b)
a = b*internal2(b)
end

function a = internal2(b)
a = b*3
end
```
If you do it like that, everything gets formatted as you would expect it...

However if the `end`-tag is not neccessary in matlab I will add correct formatting for this.
Then I will need to know exactly when `end `is not needed. A little bit of assistance would be appreciated :)

**Comment by FYPetro** [comment]
In the example I provided, the internal functions `internal` and `internal2` don't necessarily have to invoke each other, so from the eyes of MATLAB they both are *internal functions*.

From [the MATLAB documentation](https://www.mathworks.com/help/matlab/matlab_prog/nested-functions.html):

> Typically, functions do not require an `end` statement. However, to nest any function in a program file, *all* functions in that file must use an `end` statement.

It seems to say that if a function `f` is defined within an `function ... end` block, `f` is called *nested* instead of internal, so nested functions can have successively bigger indent. I believe this means that all internal functions should rest on the same level of indent.

**Comment by affenwiesel** [comment]
Well, this might be quite tricky to implement. In my current code I am parsing the file line by line. So there is no way of knowing if a function is nested or if the programmer just chose not to use an end-statement. 
Also as long as one is always using `end` (which is good practice anyway imo) everything gets formatted correctly. So I don't really see a pressing need to implement such a feature...
Long story short: I'm not going to do anything about it but feel free to contribute it yourself.
